### ISSUE TITLE
Higher score in CoNLL-2014 test by inference tweaks than paper

### ISSUE THREAD

**Issue by KYRIEZX** [grammarly/gector#65]
I got the following score by tweaking Confidence Bias and Min-Error-Probability of the XLNet Gector model in CoNLL-2014:
Precision: 0.8031, Recall: 0.4220, F_0.5: 0.6802
It's higher than the score you show in the paper which is 0.653 of F_0.5. I noticed that you say "We forced the model to perform more precise corrections by introduc- ing two inference hyperparameters (see Appendix, Table 11), hyperparameter values were found by random search on BEA-dev". Do you mean you didn't get the fix hyperparameters to get the highest score in the experiments and the model could be even better by tweaking these two hyperparameters?

**Comment by skurzhanskyi** [comment]
Usually, inference tweaks have to balance between precision and recall. In your case, both precision and recall are higher than we reported. Can you please share Confidence Bias and Min-Error-Probability you used?

**Comment by KYRIEZX** [comment]
> Usually, inference tweaks help to balance between precision and recall. In your case, both precision and recall are higher than we reported. Can you please share Confidence Bias and Min-Error-Probability you used?

Absolutely yes. I just use the  hyperparameters you post on the Appendix Table 11 in your paper, which are 0.35 and 0.66 respectively, but you use them for the BEA-dev. Please tell me if there's anything wrong in this result!

**Comment by skurzhanskyi** [comment]
We use these parameters for both CoNLL-2014 and BEA. It was just tuned on BEA-dev.
I think there are some differences in the evaluation set. Did you use [this data](https://www.cl.cam.ac.uk/research/nl/bea2019st/#data)?

**Comment by KYRIEZX** [comment]
> We use these parameters for both CoNLL-2014 and BEA. It was just tuned on BEA-dev.
> I think there are some differences in the evaluation set. Did you use [this data](https://www.cl.cam.ac.uk/research/nl/bea2019st/#data)?

The CoNLL-2014 I used is [this](https://www.comp.nus.edu.sg/~nlp/conll14st/conll14st-test-data.tar.gz) "combined with alt" where I downloaded from [this page](http://nlpprogress.com/english/grammatical_error_correction.html).

**Comment by skurzhanskyi** [comment]
Yes, that may be the reason why you have different results. Please check the scores on the CoNLL-2014 evalset from the link I sent before.

**Comment by KYRIEZX** [comment]
> Yes, that may be the reason why you have different results. Please check the scores on the CoNLL-2014 evalset from the link I sent before.

I think so.. But the page you shared me is the BEA, isn't it?

**Comment by skurzhanskyi** [comment]
It's the link for the BEA competition, not the BEA evalsets. There're links for all the train data.
![image](https://user-images.githubusercontent.com/17638837/96723684-87031080-13b7-11eb-9e04-a29263d59224.png)
But I don't see the links for CoNLL test set 🤔

**Comment by skurzhanskyi** [comment]
Oh, got it. We use `noalt` version, as it's the official version and everyone uses it.

**Comment by KYRIEZX** [comment]
> Oh, got it. We use `noalt` version, as it's the official version and everyone uses 
Thank you! I just restarted the experiment and got the same result in the paper. So you didn't get the absolute hyperparameters for the best performance while the 0.35 and 0.66 are the best among several random tests, right?

**Comment by KYRIEZX** [comment]
> Oh, got it. We use `noalt` version, as it's the official version and everyone uses it.

Thank you! I just restarted the experiment and got the same result in the paper. So you didn't get the absolute hyperparameters for the best performance while the 0.35 and 0.66 are the best among several random tests, right?

**Comment by skurzhanskyi** [comment]
We didn't tune these parameters on CoNLL-test, cause in this case, we would overfit on this evalset.

**Comment by KYRIEZX** [comment]
> We didn't tune these parameters on CoNLL-test, cause in this case, we would overfit on this evalset.

OK, thanks. Your model is awesome!

**Comment by skurzhanskyi** [comment]
Thanks :)

**Comment by khatrimann** [comment]
Hey, when we use errant for creation of m2 file using parallel corpus, we get edits with corrections like ```A 1 1|||M:PUNCT|||,|||REQUIRED|||-NONE-|||0``` but in CoNLL we have annotations like this ```A 1 1|||Trans|||,|||REQUIRED|||-NONE-|||0```. So will there be any problem while using ```errant_compare```. And how can we switch between both the formats?
